credentials/stscreds/web_identity_provider.go

```diff
@@ -145,6 +152,14 @@ func (p *WebIdentityRoleProvider) Retrieve(ctx context.Context) (aws.Credentials
 		Source:          WebIdentityProviderName,
 		CanExpire:       true,
 		Expires:         *resp.Credentials.Expiration,
+		AccountID:       accountID,
 	}
 	return value, nil
 }
+
+func getAccountID(assumedRoleUser *types.AssumedRoleUser) string {
+	if arn := assumedRoleUser.Arn; arn != nil && len(*arn) > 0 {
+		return strings.Split(*arn, ":")[4]
+	}
+	return ""
+}
```

Comment:
You're still liable to panic on out-of-bounds this way, I'd restructure:

```suggestion
func getAccountID(u *types.AssumedRoleUser) string {
	if u.Arn == nil {
		return ""
	}

	parts := strings.Split(*u.Arn)
	if len(parts) < 5 {
		return ""
	}

	return parts[4]
}
```

Reply:
Okay I thought the arn format would be validated before the resolving. Will change that